data_process.py

```diff
@@ -406,7 +411,96 @@ def augment(data, augment_func=add_noise_to_data_point, augment_proportion=0.5,
     
     return
 
+
+def get_specific_date_data(date_data, data, start_date='2000-01-01', end_date='2020-01-01'):
+    '''
+    Parameters
+    ----------
+    date_data : numpy array dtype '<U10'
+        shape (N,1)
+    data : numpy array dtype 'float32'
+        shape (N,M)
+    start_date : string date 'YYYY-mm-dd'
+    end_date : string date 'YYYY-mm-dd'
+
+    Returns
+    -------
+    data_in_range : list of tuples (x,t)
+        x - shape (x_length, M)
+        t - shape (t_length, M)
```

Comment:
We can change the return type here to - "array shape (Q,M) - timesteps within date range" or similar